src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -337,6 +338,18 @@ trait PodsValidation {
     pod.scaling is optional(scalingValidator)
     pod is endpointNamesUnique and endpointContainerPortsUnique and endpointHostPortsUnique
   }
+
+  // Marathon 1.4.0 did not introduce a validation rule for container names.
+  // Introducing this rule needs to allow existing (possibly wrong) container names.
+  // A migration can not fix this problem, since the task id is generated from the container name
+  // and there might be started tasks.
+  // In a later version this validation rule should be applied in the default validator.
+  val validPodContainerUpdate: Validator[MesosContainer] = validator[MesosContainer] { container =>
+    container.name should matchRegexFully("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  }
+  val validPodUpdate: Validator[PodDefinition] = validator[PodDefinition] { pod =>
```

Comment:
It is a serious bug to NOT validate the container name.
Since pods is still experimental, I would not maintain this bug, but fix it.
Please note: pods with invalid container names will never run successfully in 1.4.2.
The validation is put on purpose to this place, since it will validate only pods that get changed, but not existing ones. 

For 1.5 we have a validation as part of the raml parser.
So we do not need a patch for HEAD from my point of view.